knowledge-base/datagrid-selection-column.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Implement Selection Column for DataGrid
+description: Learn how to add a selection column to the DataGrid in .NET MAUI so the user can select the entire row using or select all rows.
+type: how-to
+page_title: How to Create a Selection Column For Row selection in the RadDataGrid for .NET MAUI
+slug: datagrid-selection-column
+tags: datagrid, .net maui, selection, row, checkbox
+res_type: kb
+---
+
+## Environment
+
+| Version | Product | Author | 
+| --- | --- | ---- | 
+| 7.1.0 | Telerik UI for .NET MAUI DataGrid | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova)| 
+
+## Description
+
+In a .NET MAUI application, you want to select the row in the DataGrid through a column and select all rows from the header of this column.
+
+## Solution
+
+To implement such column and to avoid the usage of the templates, you have to use the [CellRenderer]({%slug datagrid-render-mode%}) approach. This approach uses SkiaSharp to draw the elements inside the DataGrid cell. For the scenario we are going to use a `DataGridBooleanColumn` with a Skia `CellRenderer`. The cell renderer will draw a checkbox element in the cell. Here is the approach:
```

Comment:
```suggestion
To implement such a column and to avoid the usage of the templates, use the [Cell Renderer]({%slug datagrid-render-mode%}) approach. This approach uses SkiaSharp to draw the elements inside the DataGrid cell. In such scenarios, use a `DataGridBooleanColumn` with a Skia `CellRenderer`. The cell renderer will draw a checkbox element in the cell.

Here is the implementation:
```